src/main/java/sa/assertj/iterables/ContainsAll.java

```diff
@@ -0,0 +1,33 @@
+package sa.assertj.iterables;
+
+import sa.assertj.Experiment;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ContainsAll extends Experiment {
+    static Experiment.DataProvider provider = (size, numOfSamples) -> {
+        Object[][] result = new Object[numOfSamples][];
+        for (int i = 0; i < numOfSamples; i++) {
+            ArrayList<Integer> list1 = new ArrayList<>();
+            ArrayList<Integer> list2 = new ArrayList<>();
+            Random rand = new Random();
+            int element = rand.nextInt();
+            list1.add(element);
+            list2.add(element);
+            for (int j = 1; j < size; j++) {
+                list1.add(rand.nextInt());
+            }
```

Comment:
Yeah, this should defos be a larger and shuffled list so that we don't always exit after the first one.

Maybe an option is to have something like this (we need to figure out how much we wanna scale the smaller one to the larger one though, maybe that should also be random):

```suggestion
            ArrayList<Integer> list1 = new ArrayList<>();
            ArrayList<Integer> list2 = new ArrayList<>();
            Random rand = new Random();
            
            for (int j = 1; j < rand.nextInt(size / 2); j++) {
                int element = rand.nextInt();
                list1.add(element);
                list2.add(element);
            }
            for (int j = 1; j < rand.nextInt(size / 2); j++) {
                list1.add(rand.nextInt());
            }
            Collections.shuffle(list1);
            Collections.shuffle(list2);
```